website/docs: stages cleanup (#21558)
goauthentik/authentik: website/docs/add-secure-apps/flows-stages/stages/identification/index.md

---
title: Identification stage
---

The Identification stage is the primary user-identification step in most authentik login flows.

## Overview

> This stage lets the user identify themselves by username, email address, UPN, or external source. It can also embed password entry, CAPTCHA verification, and passkey autofill directly into the identification step.

Reviewer: ```suggestion
This stage lets the user identify themselves by username, email address, UPN, or external source. You can also embed password entry, CAPTCHA verification, and passkey autofill directly into the identification step.
```

If you set a **Captcha stage** here, configure that CAPTCHA for invisible or background use so it renders correctly inside the identification form.

### Pretend user exists

When enabled, invalid identifiers still let the flow continue as long as the format is valid for the selected field type. Stages such as [Password](../password/index.md) and [Email](../email/index.md) are aware of this behavior and handle the synthetic pending user safely.

### Source behavior

Selected OAuth and SAML sources are shown below the local identification form. If no user fields are selected and exactly one source is configured, authentik can redirect to that source automatically as long as **Passwordless flow** is not configured.

By default, sources are shown by icon only. Enable **Show source labels** if the source name should be shown alongside the icon.

### Linked flows

The **Enrollment flow** and **Recovery flow** settings control the links shown below the identification form. The passwordless flow setting adds a direct link to a separate passwordless authentication flow.

> When **Enable remember me on this device** is enabled, authentik can store the username locally and fast-forward to the password field on later visits. Users can still select **Not you?** to enter a different username or disable the remembered value.

Reviewer: ```suggestion
When **Enable remember me on this device** is enabled, authentik can store the username locally and jump to the password field on later visits. Users can still select **Not you?** to enter a different username or disable the remembered value.
```

### Passkey autofill (WebAuthn conditional UI):ak-version[2025.12]

When configured, the Identification stage can offer passkey login directly from the browser's passkey or autofill UI without requiring the user to type a username first.

#### Requirements

- **HTTPS** is required except on `localhost`.
- The browser must support WebAuthn conditional mediation.
- The user must have a compatible discoverable credential.
- The user must access authentik on the same hostname the passkey was created for.

#### Configuration

1. Create or edit an [Authenticator Validation stage](../authenticator_validate/index.md) that allows the **WebAuthn** device class.
2. Set the Identification stage's **WebAuthn Authenticator Validation stage** to that stage.

> 3. Make sure users have already enrolled a WebAuthn device, for example with the [WebAuthn / FIDO2 / Passkeys Authenticator setup stage](../authenticator_webauthn/index.md).

Reviewer: ```suggestion
3. Make sure users have already enrolled a WebAuthn authenticator, for example with the [WebAuthn / FIDO2 / Passkeys Authenticator setup stage](../authenticator_webauthn/index.md).
```

If the user has multiple passkeys, the browser shows its own picker. In the default authentication flow, authentik skips the MFA validation stage after a passkey login with an expression policy; adjust that policy if you still want a second factor after passkey login.

#### Troubleshooting

- If no passkey prompt appears, check HTTPS, browser support, and that the validation stage is configured. The browser normally triggers the passkey suggestion when the user focuses the username field.

> - If the prompt appears but falls back to username/password, check that the referenced validation stage allows WebAuthn and that the user has a valid enrolled device.

Reviewer: ```suggestion
- If the prompt appears but falls back to username/password, check that the referenced validation stage allows WebAuthn and that the user has a valid enrolled authenticator.
```
